### DISCUSSION TITLE
What has to be done to upload to shared hosting?

### DISCUSSION THREAD

**Discussion by rowild** [directus/directus#17768] [Archive]
I would like to host my Directus project (using SQLite) on my provider, who concentrates on the apache/php combination and does not provide any options for container uploads nor Node.js. However, since I do have ssh access, I could manage to install at least Node.js, but my attempts to install Directus on my host haven't been successful so far, it seems a timeout is triggered (Node 16, 18, 19).

That's why I would like to upload my local instance to my shared host and I was hoping you could help me with these questions:

1. Do I have to trigger some kind of build function? Usually node projects have various options to be started like "npm run dev" or "npm build" etc. From Directus I only know the command "npx directus start".
2. Is it at all possible to upload a local installation to a remote host? I wonder because the node_modules folder is rather large, and there is always the question of links. 
3. I assume I have to adapt my .env file. Is there a way to have different .env files for local and production?
4. Anything else I am missing here?

I read on learningdirectus.com somewhere that it is not recommended to host Directus on shared servers due to power consumption. However, I have only a small website that collects mp3s and pdfs for downloads, and I do not expect any serious traffic. I would like to give it a try and see if my hosting provider (all-inkl.com, btw) can manage that. (It is a host that does support the php-based CMS TYPO3, which is also very poser consuming.)

Thank you!

**Comment by rowild** [comment]
Does nobody use Directus with shared hostings? I tried with FTP (which takes a lot of time considering the huge node_modules folder). But to no avail.

**Reply by lukas-schaetzle** [reply]
There are some people using directus on shared hosting plans (for example see https://github.com/directus/directus/discussions/15917#discussioncomment-3834275) but it is generally not recommended, not because of power consumption but often arising problems with server configurations you cannot influence without root privileges. My shared hosting plan has node.js preinstalled so I have not much experience with your problems but I can try to help:
1. No build or initialzation command needed (provided that you also copy the initialized database to the server and use the same Directus version). You can start Directus by just executing `node_modules/directus/dist/start.js` or using the mentionend npx command.
2. Theoretically yes.  However some of the dependencies are compiled on the system they are installed to. So it won't work if you copy your `node_modules` folder from for example a windows machine to a linux server. I suspect that is a problem you are running into.
3. You have to make sure to use the right port and `localhost` as Host. The rest of your local .env file should be fine.
4. I can't imagine that your hosting provider is too happy about it if you have a continuously running individual application (node.js) on a shared host. It may be shutdown after some time.

All in all, I doubt that you get satisfying results with this hosting plan. Updating Directus would also be a pain because you have to redo everything. I suggest using at least a shared host with node.js support (this usually comes with enough problems to tackle 😆)

**Reply by rijkvanzanten** [reply]
I strongly discourage using shared-hosts for hosting Directus, to @Tummerhore's credit it's pain and suffering until it kinda works, and then every update after that is more pain and suffering. I'd recommend checking out a PaaS as a replacement for that dirt-cheap low-scale usage (like DO App Platform or Railway etc)

**Reply by rowild** [reply]
@Tummerhore & @rijkvanzanten Thanks for your feedbacks! Moved on to DO quite some time ago and there everything works like a charm!
Thanks for pointing out Railway.app. Didn't know about them! :-)
